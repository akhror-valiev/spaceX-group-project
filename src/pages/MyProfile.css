.reserve-rockets {
  display: flex;
  width: 100%;
  gap: 2rem;
}
