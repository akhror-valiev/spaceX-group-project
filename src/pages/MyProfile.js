import { useSelector } from 'react-redux';
import ReadyFlyRockets from '../components/ReadyFlyRockets/ReadyFlyRockets';
import JoinMissions from '../components/JoinMission/JoinMissions';
import './MyProfile.css';

const MyProfile = () => {
  const rockets = useSelector((state) => state.rockReducer);
  const missions = useSelector((state) => state.missions);

  const readyFlyRockets = rockets.filter((rocket) => rocket.reserved === true);
  const joinedMissions = missions.filter((mission) => mission.reserved === true);

  return (
    <div className="reserve-rockets">
      <ReadyFlyRockets readyRockets={readyFlyRockets} object="Rockets" />
      <JoinMissions joinedMissions={joinedMissions} object="Mission" />
    </div>
  );
};

export default MyProfile;
