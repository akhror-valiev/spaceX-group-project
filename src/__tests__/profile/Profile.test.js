import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import MyProfile from '../../pages/MyProfile';

describe('snapshot testing', () => {
  test('snapshot for Profile component', () => {
    const rendererComponent = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>,
      )
      .toJSON();
    expect(rendererComponent).toMatchSnapshot();
  });
});
