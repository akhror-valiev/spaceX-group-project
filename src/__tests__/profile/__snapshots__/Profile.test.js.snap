// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for Profile component 1`] = `
<div
  className="reserve-rockets"
>
  <div
    className="container"
  >
    <h1>
      My
      Rockets
    </h1>
    <p
      className="no-reserve"
    >
      No Reservation
    </p>
  </div>
  <div
    className="container"
  >
    <h1>
      My
      Mission
    </h1>
    <p
      className="no-reserve"
    >
      No Mission Joined
    </p>
  </div>
</div>
`;
